flex-squid/src/main/java/org/sonar/flex/lexer/FlexLexer.java

```diff
@@ -43,36 +44,49 @@ private FlexLexer() {
 
   public static Lexer create(FlexConfiguration conf) {
     return Lexer.builder()
-        .withCharset(conf.getCharset())
+      .withCharset(conf.getCharset())
 
-        .withFailIfNoChannelToConsumeOneCharacter(true)
+      .withFailIfNoChannelToConsumeOneCharacter(true)
 
-        .withChannel(new BomCharacterChannel())
-        .withChannel(new BlackHoleChannel("\\s++"))
+      .withChannel(new BomCharacterChannel())
+      .withChannel(new BlackHoleChannel("\\s++"))
 
-        // Comments
-        .withChannel(commentRegexp("//[^\\n\\r]*+"))
-        .withChannel(commentRegexp("/\\*[\\s\\S]*?\\*/"))
+      // Comments
+      .withChannel(commentRegexp("//[^\\n\\r]*+"))
+      .withChannel(commentRegexp("/\\*[\\s\\S]*?\\*/"))
 
-        // String Literals
-        .withChannel(regexp(GenericTokenType.LITERAL, "\"([^\"\\\\]*+(\\\\[\\s\\S])?+)*+\""))
-        .withChannel(regexp(GenericTokenType.LITERAL, "\'([^\'\\\\]*+(\\\\[\\s\\S])?+)*+\'"))
+      // String Literals
+      .withChannel(regexp(GenericTokenType.LITERAL, "\"([^\"\\\\]*+(\\\\[\\s\\S])?+)*+\""))
+      .withChannel(regexp(GenericTokenType.LITERAL, "\'([^\'\\\\]*+(\\\\[\\s\\S])?+)*+\'"))
 
-        // Regular Expression Literal
-        .withChannel(new FlexRegularExpressionLiteralChannel())
+      // Regular Expression Literal
+      .withChannel(new FlexRegularExpressionLiteralChannel())
 
-        // Numbers
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "0[xX][0-9a-fA-F]++"))
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++\\.([0-9]++)?+" + EXP + "?+"))
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "\\.[0-9]++" + EXP + "?+"))
-        .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++" + EXP + "?+"))
+      // Numbers
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "0[xX][0-9a-fA-F]++"))
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++\\.([0-9]++)?+" + EXP + "?+"))
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "\\.[0-9]++" + EXP + "?+"))
+      .withChannel(regexp(FlexTokenType.NUMERIC_LITERAL, "[0-9]++" + EXP + "?+"))
 
-        .withChannel(new IdentifierAndKeywordChannel("\\p{javaJavaIdentifierStart}++\\p{javaJavaIdentifierPart}*+", true, FlexKeyword.values()))
-        .withChannel(new PunctuatorChannel(FlexPunctuator.values()))
+      .withChannel(new IdentifierAndKeywordChannel("\\p{javaJavaIdentifierStart}++\\p{javaJavaIdentifierPart}*+", true, FlexKeyword.values()))
+      .withChannel(new PunctuatorChannel(FlexPunctuator.values()))
 
-        .withChannel(new UnknownCharacterChannel())
+      .withChannel(new UnknownCharacterChannel())
 
-        .build();
+      .build();
+  }
+
+  public static class BomCharacterChannel extends Channel<Lexer> {
+
+    public boolean consume(CodeReader code, Lexer lexer) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1161)